File: e2e/playwright-typescript/interactions/steps/CreateTaskStep.ts
import { expect, Page } from '@playwright/test';
import { step } from '@/decorators/step-decorator';
import { CreateTaskForm } from '@/interactions/pages/CreateTaskForm';
import { TaskData } from '@/data/TaskData';

export class CreateTaskStep {
  private readonly form: CreateTaskForm;

  constructor(page: Page) {
    this.form = new CreateTaskForm(page);
  }

  @step('Fill create task form with {taskData}')
  async fillForm(taskData: TaskData): Promise<void> {
    await this.form.titleInput.fill(taskData.title);
    await this.form.descriptionInput.fill(taskData.description);
    await this.form.statusDropdown.click();
    await this.form.statusOption(taskData.status).click();
    await this.form.priorityDropdown.click();
    await this.form.priorityOption(taskData.priority).click();
  }

  @step('Submit create task form')
  async submitForm(): Promise<void> {
    await this.form.submitButton.click();
    await expect(this.form.submitButton).not.toBeVisible();
  }
}

File: e2e/playwright-python/interactions/steps/create_task_step.py
from __future__ import annotations

from playwright.sync_api import Page, expect
from data.task_data import TaskData
from interactions.pages.create_task_form import CreateTaskForm
from decorators.step_decorator import step


class CreateTaskStep:
    def __init__(self, page: Page):
        self._form = CreateTaskForm(page)

    @step("Set task data with {task_data}")
    def set_task_data(self, task_data: TaskData) -> CreateTaskStep:
        self._form.title_input.fill(task_data.title)
        self._form.description_input.fill(task_data.description)
        self._form.status_dropdown.click()
        self._form.status_option(task_data.status).click()
        self._form.priority_dropdown.click()
        self._form.priority_option(task_data.priority).click()
        return self

    @step("Submit create task form")
    def submit_form(self) -> None:
        self._form.submit_button.click()
        expect(self._form.submit_button).not_to_be_visible()
